### DISCUSSION TITLE
How to build go binary for cross platforms

### DISCUSSION THREAD

**Discussion by r0ckysec** [projectdiscovery/naabu#278] [Q&A]
I would like to ask how Mac M1 ARM compiles and generates linux amd64

```
GOOS=linux GOARCH=amd64 CGO_ENABLED=1 CC=x86_64-linux-musl-gcc CGO_LDFLAGS="-static" go build -a
```
![image](https://user-images.githubusercontent.com/44518337/156807323-a084f3ae-ace4-4fe7-80c9-8e0b0d8440e3.png)

**Comment by ehsandeep** [comment]
@r0ckysec not sure what you mean by that, but you need to have `libpcap` installed on system before installing naabu as [mentioned](https://github.com/projectdiscovery/naabu#installation-instructions) in readme.

**Reply by r0ckysec** [reply]
The current system is Apple M1 Max Arm64，Built-in `libpcap`
I now want to develop on mac, cross-compile into linux amd64 binaries, which can run on the server area.
How to cross compile cgo on mac to generate linux amd64

**Reply by ehsandeep** [reply]
https://github.com/goreleaser/goreleaser

**Reply by r0ckysec** [reply]
I have used this and searched for relevant information, but I don't know what parameters to configure to cross compile cgo.
